extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt

```diff
@@ -0,0 +1,192 @@
+package com.mapbox.maps.extension.androidauto
+
+import android.graphics.Rect
+import com.mapbox.common.Logger
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.ScreenCoordinate
+import com.mapbox.maps.plugin.animation.camera
+import java.util.concurrent.CopyOnWriteArraySet
+
+/**
+ * @see MapboxCarMap to create new map experiences.
+ *
+ * Maintains the surface state for [MapboxCarMap].
+ */
+@OptIn(MapboxExperimental::class)
+internal class CarMapSurfaceOwner {
+
+  internal var mapboxCarMapSurface: MapboxCarMapSurface? = null
+    private set
+  internal var visibleArea: Rect? = null
+    private set
+  internal var edgeInsets: EdgeInsets? = null
+    private set
+  internal var visibleCenter: ScreenCoordinate = visibleCenter()
+    private set
+
+  private val carMapObservers = CopyOnWriteArraySet<MapboxCarMapObserver>()
+
+  fun registerObserver(mapboxCarMapObserver: MapboxCarMapObserver) {
+    carMapObservers.add(mapboxCarMapObserver)
+    Logger.i(TAG, "registerObserver + 1 = ${carMapObservers.size}")
+
+    mapboxCarMapSurface?.let { carMapSurface ->
+      mapboxCarMapObserver.onAttached(carMapSurface)
+    }
+    ifNonNull(mapboxCarMapSurface, visibleArea, edgeInsets) { _, area, edge ->
+      Logger.i(TAG, "registerObserver visibleAreaChanged")
+      mapboxCarMapObserver.onVisibleAreaChanged(area, edge)
+    }
+  }
+
+  fun unregisterObserver(mapboxCarMapObserver: MapboxCarMapObserver) {
+    carMapObservers.remove(mapboxCarMapObserver)
+    mapboxCarMapSurface?.let { mapboxCarMapObserver.onDetached(it) }
+    Logger.i(TAG, "unregisterObserver - 1 = ${carMapObservers.size}")
+  }
+
+  fun clearObservers() {
+    val oldCarMapSurface = this.mapboxCarMapSurface
+    oldCarMapSurface?.let { carMapObservers.forEach { it.onDetached(oldCarMapSurface) } }
```

Comment:
Just 
```
this.mapboxCarMapSurface?.let { carMapObservers.forEach { it.onDetached(it) } }
```
?

Reply:
yeah simpler and no functional difference. updating